test/xbmcgui.py

```diff
@@ -44,6 +44,12 @@ def textviewer(heading, text=None, usemono=None):
         text = kodi_to_ansi(text)
         print('\033[37;100mTEXTVIEWER:\033[35;0m [%s]\n\033[35;0m%s\033[39;0m' % (heading, text))
 
+    @staticmethod
+    def browseSingle(atype, heading, shares, mask=None, useThumbs=None, treatAsFolder=None, default=None):
+        ''' A stub implementation for the xbmcgui Dialog class browseSingle() method '''
+        print('[37;100mBROWSESINGLE:[35;0m [%s] [35;0m%s[0m' % (atype, heading))
+        return '~/.kodi/userdata/addon_data/script.module.inputstreamhelper'
```

Comment:
I would probably return a special path (the default one).

Reply:
BTW You can turn this back into `type`. With the new xbmc framework update I have ignored the `invalid-name` checks for these modules. (This is in fact an issue with the original modules :disappointed:)